### ISSUE TITLE
Rust does not warn about impossible conditions [feature request]

### ISSUE THREAD

**Issue by amarao** [rust-lang/rust#52974]
Rust silently compiles conditions which are always false, without emitting any warnings.

Example:
```
fn main() {
    let a = 1;
    if a < 1 && a > 1 {
        println!("Impossible");
    }
}
```
I think it should warn about conditions which are always true or always false, because such conditions may be a human mistake (wrong variable name, wrong operator, wrong parentheses, etc).

**Comment by oli-obk** [comment]
This feels like it belongs into clippy. We already have lints in this direction

**Comment by dtolnay** [comment]
Let's track this in Clippy as part of https://github.com/rust-lang-nursery/rust-clippy/issues/2054.
